### ISSUE TITLE
Scanner is active in background when Navigating to another screen

### ISSUE THREAD

**Issue by arnabonetraker** [juliansteenbakker/mobile_scanner#619]
My flow requires when barcode is detected navigate to new screen, do some activity then pop to scanner screen again.
Now the issue is the scanner is active in background, so it is detecting the barcode again even when I am in a different screen.

` _mobileScannerController = MobileScannerController(facing: CameraFacing.back,
      formats: [BarcodeFormat.code128],
      detectionSpeed: DetectionSpeed.normal,
      autoStart: true,
    )`
    This is my controller settings. I have tried stoping the camera using `stop()` before navigating to another screen.
    then using callback to `start()` again when it is poped.
    This works for android but it is not working in a IOS device. For IOS the scanner is still activate in the background.
    
```flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.12, on macOS 13.3.1 22E772610a darwin-x64, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

**Comment by teolemon** [comment]
- Is it related to https://github.com/juliansteenbakker/mobile_scanner/issues/589 ?

**Comment by arnabonetraker** [comment]
> * Is it related to [Camera does not work after navigation from one screen to another (Both use MobileScanner) #589](https://github.com/juliansteenbakker/mobile_scanner/issues/589) ?

Not quite. In my case scanner keep detecting barcode even while navigating to other screen(no scanner initialised in this second screen).

**Comment by denihero** [comment]
In my case after navigation I do `cameraController.stop(),` but when I return to scan page camera is not working.

**Comment by arnabonetraker** [comment]
> In my case after navigation I do `cameraController.stop(),` but when I return to scan page camera is not working.

did you set your `MobileScannerController(autoStart:true)` ?
If not then you will have to `cameraController.start()` when you pop and get the callback.

**Comment by denihero** [comment]
> > In my case after navigation I do `cameraController.stop(),` but when I return to scan page camera is not working.
> 
> did you set your `MobileScannerController(autoStart:true)` ? If not then you will have to `cameraController.start()` when you pop and get the callback.

`autoStart:true` stau automaticly, I tried do pass controller to new Screen and call `cameraController.start()`, but it's not working

**Comment by JoseSarricolea** [comment]
@denihero 
I think the problem is in the controller.
I managed, Overwriting the controller every time it wants to scan again. Try this:

Declare the controller so you don't have to check for null in functions:
`MobileScannerController cameraController = MobileScannerController();`

When I want to stop the scanner:

` FutureOr<void> _onQrViewCameraStopped(
      QrViewCameraStopped event, Emitter<QrViewState> emit) {
    cameraController.stop();
    cameraController.dispose();
  }`

When I want to scan a qr:
`FutureOr<void> _onQrViewLoaded(
      QrViewLoaded event, Emitter<QrViewState> emit) {
    cameraController = MobileScannerController();
    emit(QrViewLoad(event.plants));
  }`

**Comment by denihero** [comment]
> @denihero I think the problem is in the controller. I managed, Overwriting the controller every time it wants to scan again. Try this:
> 
> Declare the controller so you don't have to check for null in functions: `MobileScannerController cameraController = MobileScannerController();`
> 
> When I want to stop the scanner:
> 
> ` FutureOr<void> _onQrViewCameraStopped( QrViewCameraStopped event, Emitter<QrViewState> emit) { cameraController.stop(); cameraController.dispose(); }`
> 
> When I want to scan a qr: `FutureOr<void> _onQrViewLoaded( QrViewLoaded event, Emitter<QrViewState> emit) { cameraController = MobileScannerController(); emit(QrViewLoad(event.plants)); }`

Where did you get `QrViewLoaded `and `QrViewState `?

**Comment by JoseSarricolea** [comment]
Am using Bloc on my proyect

**Comment by braveshine1993m** [comment]
In case people still dealing with this issue, you can do like below :
```dart
controller.stop();
Navigator.pushNamed(context, 'someRoute').then((value) {
    controller.start();
});
```

if this wasn't enough, do : `Navigator.pop(context, true)` in second page.
No need to bloc or etc.

**Comment by denihero** [comment]
> 

thanks, man, it's working, I'll appreciate it

**Comment by michael11albrecht** [comment]
So as I see at the moment there is no option to pop back from the Scanner page and later use the Scanner again?
I tried different approaches, stopping and starting the controller manually or completely dispose the controller.
But I couldn't find a way getting the Scanner run again after leaving it ones (except leaving it forward (push) and afterwards going back like @braveshine1993m recommendet.
I tested on an iPhone 12 ios 16.5.1 and the android emulator.
